examples/sphDiagramGeneration/sph-diagram-generation.cpp

```diff
@@ -132,82 +100,24 @@ int main(int argc, char *argv[]) {
 
-  for (auto it = lcCont.begin(); it.isValid(); ++it) {
-    dirCont.addParticle(*it);
-    verletCont.addParticle(*it);
-    verletCellCont.addParticle(*it);
-  }
-
-  if (containerType == linkedCells) {
-    if (functorType == densityFunctor) {
-      measureContainer(&lcCont, &densfunc, numParticles, numIterations, useNewton3);
-    } else if (functorType == hydroForceFunctor) {
-      measureContainer(&lcCont, &hydrofunc, numParticles, numIterations, useNewton3);
-    } else {
-      std::cout << "wrong functor given" << std::endl;
-      exit(2);
-    }
-  } else if (containerType == directSum) {
-    if (functorType == densityFunctor) {
-      measureContainer(&dirCont, &densfunc, numParticles, numIterations, useNewton3);
-    } else if (functorType == hydroForceFunctor) {
-      measureContainer(&dirCont, &hydrofunc, numParticles, numIterations, useNewton3);
-    } else {
-      std::cout << "wrong functor given" << std::endl;
-      exit(2);
-    }
-  } else if (containerType == verletLists) {
-    if (functorType == densityFunctor) {
-      measureContainer(&verletCont, &densfunc, numParticles, numIterations, useNewton3);
-    } else if (functorType == hydroForceFunctor) {
-      measureContainer(&verletCont, &hydrofunc, numParticles, numIterations, useNewton3);
-    } else {
-      std::cout << "wrong functor given" << std::endl;
-      exit(2);
-    }
-  } else if (containerType == verletListsCells) {
-    if (functorType == densityFunctor) {
-      measureContainer(&verletCellCont, &densfunc, numParticles, numIterations, useNewton3);
-    } else if (functorType == hydroForceFunctor) {
-      measureContainer(&verletCellCont, &hydrofunc, numParticles, numIterations, useNewton3);
-    } else {
-      std::cout << "wrong functor given" << std::endl;
-      exit(2);
-    }
-  } else {
-    std::cout << "invalid container id" << std::endl;
-    exit(3);
-  }
-}
+  autoPas.setBoxMin(boxMin);
+  autoPas.setBoxMax(boxMax);
+  autoPas.setCutoff(cutoff);
+  autoPas.setVerletSkin(skin * cutoff);
+  autoPas.setAllowedContainers(containerOptions);
+  autoPas.setAllowedNewton3Options({useNewton3 ? autopas::Newton3Option::enabled : autopas::Newton3Option::disabled});
+  autoPas.setAllowedDataLayouts({autopas::DataLayoutOption::aos});  // currently aos only!
 
-template <class Container, class Functor>
-void measureContainer(Container *cont, Functor *func, int numParticles, int numIterations, bool useNewton3) {
-  using CellType = autopas::FullParticleCell<autopas::sph::SPHParticle>;
-  // initialize to dummy values
-  auto traversalInfo = cont->getTraversalSelectorInfo();
-  std::cout << "Cells: " << traversalInfo.dims[0] << " x " << traversalInfo.dims[1] << " x " << traversalInfo.dims[2]
-            << std::endl;
   auto traversalType = autopas::TraversalOption(-1);
-  switch (cont->getContainerType()) {
+  switch (*containerOptions.begin()) {
```

Comment:
I don't understand this `switch`. Why does it only check the first container option? Why do you only allow one traversal per container and why not let the user decide which one?

Reply:
only 1 container option is allowed ;)

Reply:
ok didn't see the if in line 76.